### ISSUE TITLE
ClientMetadata not sent in PASSWORD_VERIFIER challenge response

### ISSUE THREAD

**Issue by uliluckas** [aws-amplify/amplify-swift#2983]
### Describe the bug

When using CUSTOM_AUTH + SRP flow, the clientMetadata passed with signIn() is not sent to the server in the PASSWORD_VERIFIER challenge response.
As the PASSWORD_VERIFIER challenge is resolved transparently by the SDK, there is no way for the developer to pass clientMetadata for the first CUSTOM_AUTH challenge's CreatAuthChallenge Lambda.
I'd therefor expect the signIn() call's clientMetadata to be passed with the PASSWORD_VERIFIER challenge response, just as in the Amplify Android SDK.

This is probably due to this [line of code](https://github.com/aws-amplify/amplify-swift/blob/10525085963fd96eb1c767ebe3f8b300a6496563/AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Utils/RespondToAuthChallengeInput%2BAmplify.swift#LL32C33-L32C33).

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Set up an authentication flow with SRP followed by one or more custom challenges
2. On the client call signIn(username, password, options) with clientMetadata
3. Observe clientMetadata not being passed to the first custom challenge's CreateCustomChallenge Lambda.
```

### Expected behavior

The signIn() call's clientMetadata should be passed with the PASSWORD_VERIFIER challenge response, just as in the Amplify Android SDK.

### Amplify Framework Version

2.11.3

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

none

### Xcode version

14.2

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Device

iPhone 11

### iOS Version

16

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by uliluckas** [comment]
Hi @harsh62 , we'll test it today. Thanks.

**Comment by uliluckas** [comment]
@harsh62, seems to work fine. thanks!

**Comment by harsh62** [comment]
@uliluckas The fix has been released in [2.11.5](https://github.com/aws-amplify/amplify-swift/releases/tag/2.11.5)
